backend/src/database/services/database-service.ts

```diff
@@ -0,0 +1,40 @@
+import type { DataSource, EntityManager } from 'typeorm';
+import { DatabaseError } from '../database-error';
+import { ActionWrapperService } from './action-wrapper.service';
+import { ExerciseWrapperService } from './exercise-wrapper.service';
+
+export class DatabaseService {
+    public readonly actionWrapperService: ActionWrapperService;
+    public readonly exerciseWrapperService: ExerciseWrapperService;
+
+    public constructor(private readonly dataSource: DataSource) {
+        this.exerciseWrapperService = new ExerciseWrapperService(dataSource);
+        this.actionWrapperService = new ActionWrapperService(
+            this.exerciseWrapperService,
+            dataSource
+        );
+    }
+
+    public get isInitialized() {
+        return this.dataSource.isInitialized;
+    }
+
+    private transactionActive = false;
```

Comment:
Maybe rename it to `transactionSemaphore`?

Reply:
I'm removing the whole "only one transaction at a time" because it doesn't prevent *nested* transactions but rather *simultaneous* transactions, which is perfectly valid.